### PR TITLE
Prefix list for V4 API

### PR DESCRIPTION
**Pull request by yachtcaptain23**
(1) Generates prefix list in protobuf format
(2) Generates channel responses in protobuf format
(3) Covers differential data

### CODE REVIEW

File: protos/channel_responses.proto [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+message SiteBannerDetails {
+  string title = 1;
+  string description = 2;
+  string background_url = 3;
+  string logo_url = 4;
+  repeated int32 donation_amounts = 5;
+  SocialLinks social_links = 6;
+}
+
+enum WalletConnectedState {
```
Comment (corymcdonald) [inline protos/channel_responses.proto:21]: So right now it seems like we can only have one type of wallet connected. Is there plans to have a one account to many wallet support?

File: app/jobs/cache/browser_channels/prefix_list.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    temp_file = Tempfile.new(["all_channels", ".br"])
+    info = Brotli.deflate(result)
+    File.open(temp_file.path, 'wb') do |f|
+      f.write(info)
+    end
+    require 'aws-sdk-s3'
+    Aws.config[:credentials] = Aws::Credentials.new(Rails.application.secrets[:s3_rewards_access_key_id], Rails.application.secrets[:s3_rewards_secret_access_key])
+    s3 = Aws::S3::Resource.new(region: Rails.application.secrets[:s3_rewards_bucket_region])
+    obj = s3.bucket(Rails.application.secrets[:s3_rewards_bucket_name]).object(ALL_CHANNELS_KEY + EXTENSION)
+    obj.upload_file(temp_file.path)
```
[the remark this answers is not in the document]
Comment (yachtcaptain23) [reply app/jobs/cache/browser_channels/prefix_list.rb:33]: It's the root file path, so this should be fine. The previous line specifies where it gets saved to

File: app/models/channel.rb [hunk 3]
```diff
@@ -271,6 +287,20 @@ def register_channel_for_promo
     Promo::RegisterChannelForPromoJob.perform_now(channel_id: id, attempt_count: 0)
   end
 
+  def update_site_banner_lookup!
+    site_banner_lookup = SiteBannerLookup.find_or_initialize_by(
+      channel_identifier: details&.channel_identifier || details.brave_publisher_id,
+    )
+    site_banner_lookup.set_sha2_base16
+    site_banner_lookup.set_wallet_status(publisher: publisher)
```
[the remark this answers is not in the document]
Comment (yachtcaptain23) [reply app/models/channel.rb:295]: I agree it seems awfully specific, but my experience from building the JSON file for V3 was that minimizing joins would drastically speed things up and decrease memory use on the worker that builds the file.

File: protos/channel_responses.proto (hunk 1 above)
Comment (yachtcaptain23) [reply protos/channel_responses.proto:21]: I'll refactor this part completely

**Comment by yachtcaptain23** [comment]
Closing in favor of https://github.com/brave-intl/publishers/pull/2697

**Pull request closed without merging**